sklearn/metrics/pairwise.py

```diff
@@ -1446,12 +1446,12 @@ def _precompute_metric_params(X, Y, metric=None, **kwds):
     """
     if metric == "seuclidean" and 'V' not in kwds:
         if X is Y:
-            V = np.var(X, axis=0, ddof=1)
+            V = np.var(X, axis=0, ddof=1, dtype=np.double)
```

Comment:
Nit:
```suggestion
            V = np.var(X, axis=0, ddof=1, dtype=np.float64)
```

Is there a difference in this case?

Reply:
From memory this caused some issues on the 32-bit system tests, but I can give it another go

Reply:
@thomasjpfan I've looked into it a bit more and I'm actually a little confused about your suggestion.  According to https://numpy.org/devdocs/release/1.20.0-notes.html, np.double is not deprecated (maybe you're thinking of np.float?)

Reply:
It's not about deprecation. It's just that we usually use np.float64 instead of double and np.float32 instead of float